File: src/modules/iam/application/internal/authService.js
import { users } from "../../../users/infrastructure/persistence/memory/usersRepository.js";

export const loginUser = (email, password) => {
  const user = users.find(u => u.email === email && u.password === password);
  if (!user) return null;
  return { id: user.id, name: user.name, email: user.email, role: user.role };
};

File: LogisPe.Api/IAM/AuthService.cs
using System.Linq;
using LogisPe.Api.Shared;

namespace LogisPe.Api.IAM;

public class AuthService
{
    public object? Login(string email, string password)
    {
        var u = InMemoryDb.Users.FirstOrDefault(x => x.Email == email && x.Password == password);
        if (u is null) return null;
        return new { id = u.Id, name = u.Name, email = u.Email, role = u.Role };
    }
}